### PR TITLE
♻️ Remove `parseUrlDeprecated` from: amp-dynamic-css-classes, amp-story 1.0, amp-iframe, amp-form, amp-sidebar

### LINKED ISSUE #15642: amp-story: Switch from parseUrl to urlForDoc
Switch from parseUrl to urlForDoc.

Context:

> PSA to extension owners, if you're importing `parseUrl` from `src/url.js`, switch to the new urlForDoc service. It'll save you ~1kb of file size

### PR DESCRIPTION
**Pull request by alanorozco**
#- Some cleanups passing by.
- Removed usages of `vsync` where possible.

Closes #15642

- [x] `amp-dynamic-css-classes`
- [x] `amp-story` 1.0
- [x] `amp-iframe`
- [x] `amp-form`
- [x] `amp-sidebar`

Size of `v0.js` bumped to include references from URL service to `src/url` as a temporary measure before all methods are migrated into service (/cc @jridgewell).

Impact on `v0.js`:

```js
f.isProtocolValid = function(a) {
    return hb(a)
};
f.getSourceOrigin = function(a) {
    return jb(a)
};
f.isSecureUrl = function(a) {
    return db(a)
};
f.assertHttpsUrl = function(a, b, c) {
    return eb(a, b, void 0 === c ? "source" : c)
};
f.assertAbsoluteHttpOrHttpsUrl = function(a) {
    z().assert(/^https?\:/i.test(a), 'URL must start with "http://" or "https://". Invalid value: %s', a);
    return I(a).href
};
f.isProxyOrigin = function(a) {
    return gb(a)
};
```

### CODE REVIEW

File: extensions/amp-story/1.0/related-articles.js [hunk 1]
```diff
@@ -1,93 +0,0 @@
-/**
```
Comment (alanorozco) [inline extensions/amp-story/1.0/related-articles.js]: File not in use. /cc @Enriqe

File: extensions/amp-story/1.0/bookend/bookend-component.js [hunk 2]
```diff
@@ -66,25 +60,42 @@ const textBoxComponentBuilder = new TextBoxComponent();
...
  * Dispatches the components to their specific builder classes.
- * @param {string} componentType
+ * @param {string} type
  * @return {?BookendComponentClass}
+ * @restricted
  */
-function componentBuilderInstanceFor(componentType) {
-  switch (componentType) {
+function componentBuilderInstanceFor(type) {
+  switch (type) {
```
Comment (alanorozco) [inline extensions/amp-story/1.0/bookend/bookend-component.js]: Drive-by cleanup to prevent instantiating on initial story render. /cc @Enriqe

File: extensions/amp-form/0.1/amp-form.js [hunk 3]
```diff
@@ -209,8 +206,9 @@ export class AmpForm {
   getXhrUrl_(attribute) {
     const url = this.form_.getAttribute(attribute);
     if (url) {
-      assertHttpsUrl(url, this.form_, attribute);
-      user().assert(!isProxyOrigin(url),
+      const urlService = Services.urlForDoc(this.element);
```
Comment (cvializ) [inline extensions/amp-form/0.1/amp-form.js]: `AmpForm` doesn't inherit from `AMP.BaseElement` so it doesn't have a `this.element`. These need to be `this.form_`.

File: extensions/amp-iframe/0.1/amp-iframe.js [hunk 4]
```diff
@@ -135,24 +135,27 @@ export class AmpIframe extends AMP.BaseElement {
    * @private
    */
   assertSource_(src, containerSrc, sandbox = '') {
-    const url = parseUrlDeprecated(src);
+    const {element: el} = this;
```
Comment (cvializ) [inline extensions/amp-iframe/0.1/amp-iframe.js]: nit: `el` seems unnecessarily abbreviated, but it's probably a personal taste thing.

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 5]
```diff
@@ -23,15 +23,17 @@ import {closestByTag, isRTL, tryFocus} from '../../../src/dom';
 import {createCustomEvent} from '../../../src/event-helper';
 import {debounce} from '../../../src/utils/rate-limit';
 import {dev} from '../../../src/log';
-import {parseUrlDeprecated, removeFragment} from '../../../src/url';
+import {removeFragment} from '../../../src/url';
 import {setStyles, toggle} from '../../../src/style';
 import {toArray} from '../../../src/types';
-/** @const */
+
+/** @private @const {string} */
```
Comment (cvializ) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: nit: The annotation changes are unnecesary since Closure can infer from the declaration that these are `@const` and `string` or `number` primitives. Did the linter complain about these?

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 6]
```diff
@@ -373,24 +369,22 @@ export class AmpStoryBookend extends AMP.BaseElement {
...
-            FULLBLEED_CLASSNAME, state.shouldBeFullBleed);
-      },
-    }, {});
+    let shouldBeFullBleed;
+    this.measureMutateElement(() => {
+      shouldBeFullBleed =
+          this.getOverflowContainer_()./*OK*/scrollTop >= FULLBLEED_THRESHOLD;
+    }, () => {
+      this.getShadowRoot().classList.toggle(
+          FULLBLEED_CLASSNAME, !!shouldBeFullBleed);
```
Comment (cvializ) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: nit: Consider initializing it to `false` so that you don't have to coerce it from `undefined` with `!!`

Comment (alanorozco) [reply extensions/amp-story/1.0/bookend/amp-story-bookend.js]: Done.

File: extensions/amp-iframe/0.1/amp-iframe.js (hunk 4 above)
Comment (alanorozco) [reply extensions/amp-iframe/0.1/amp-iframe.js]: True, fixed.

File: extensions/amp-form/0.1/amp-form.js (hunk 3 above)
Comment (alanorozco) [reply extensions/amp-form/0.1/amp-form.js]: Yup, fixed.

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 5 above)
Comment (alanorozco) [reply extensions/amp-sidebar/0.1/amp-sidebar.js]: Just for completion.

File: extensions/amp-form/0.1/test/test-form-proxy.js [hunk 7]
```diff
@@ -33,6 +35,22 @@ describes.repeated('installFormProxy', {
 }, (name, variant) => {
   let form;
   let inputs;
+  let sandbox;
+
+  before(() => {
+    sandbox = sinon.sandbox.create();
```
Comment (aghassemi) [inline extensions/amp-form/0.1/test/test-form-proxy.js]: there shouldn't be a need to create a `sandbox` the new `describes` provides a global, already-instantiated `sandbox`

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 8]
```diff
@@ -161,16 +158,16 @@ export class AmpSidebar extends AMP.BaseElement {
...
     this.registerAction('toggle', this.toggle_.bind(this));
     this.registerAction('open', this.open_.bind(this));
     this.registerAction('close', this.close_.bind(this));
 
-    this.element.addEventListener('click', e => {
+    element.addEventListener('click', e => {
       const target = closestByTag(dev().assertElement(e.target), 'A');
       if (target && target.href) {
-        const tgtLoc = parseUrlDeprecated(target.href);
+        const tgtLoc = Services.urlForDoc(element).parse(target.href);
```
Comment (aghassemi) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: should it go to an instance variable? there are 2-3 usages in the component

File: extensions/amp-form/0.1/test/test-form-proxy.js (hunk 7 above)
Comment (alanorozco) [reply extensions/amp-form/0.1/test/test-form-proxy.js]: I get `ReferenceError` if I try to access `sandbox` globally. I don't  think `repeated` sets up the environment.

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 8 above)
Comment (alanorozco) [reply extensions/amp-sidebar/0.1/amp-sidebar.js]: `amp-sidebar` only has one. `amp-iframe` has two, added an instance variable there.

File: extensions/amp-story/1.0/bookend/bookend-component.js (hunk 2 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/bookend/bookend-component.js]: Very cool, thanks @alanorozco !

File: extensions/amp-story/1.0/related-articles.js (hunk 1 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/related-articles.js]: Woops, forgot to delete it. Thanks!!

File: extensions/amp-form/0.1/test/test-form-proxy.js (hunk 7 above)
Comment (cvializ) [reply extensions/amp-form/0.1/test/test-form-proxy.js]: this is correct

File: extensions/amp-iframe/0.1/amp-iframe.js [hunk 9]
```diff
@@ -120,6 +121,12 @@ export class AmpIframe extends AMP.BaseElement {
      * @private {?string}
      */
     this.targetOrigin_ = null;
+
+    /**
+     * @return {!../../../src/service/url-impl.Url}
+     * @private
+     */
+    this.getUrlService_ = once(() => Services.urlForDoc(this.element));
```
Comment (jridgewell) [inline extensions/amp-iframe/0.1/amp-iframe.js]: This `once` stuff isn't necessary, neither access is in the fast path.

Comment (alanorozco) [reply extensions/amp-iframe/0.1/amp-iframe.js]: Will remove in a follow-up PR.

Comment (jridgewell) [reply extensions/amp-iframe/0.1/amp-iframe.js]: [Already have one](https://github.com/ampproject/amphtml/pull/15614). 😉
